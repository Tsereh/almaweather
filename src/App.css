.App {
  text-align: center;
}

.App-logo {
  animation: App-logo-spin infinite 20s linear;
  height: 40vmin;
  pointer-events: none;
}

.App-header {
  background-color: #282c34;
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
}

.App-link {
  color: #61dafb;
}

@keyframes App-logo-spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}

.search-field {
  box-shadow: 0px 1px 5px 0px rgba(0,0,0,0.26);
}

.main-weather {
  background-color: rgba(238, 238, 238, 0.37);
  border-radius: 12px;
  box-shadow: 0px 1px 5px 0px rgba(0,0,0,0.26);
  margin-top: 10px;
  color: rgba(0,0,0,0.75);
  vertical-align: middle;
  height: 360px;
  width: 600px;
  padding-top: 114px;
}
