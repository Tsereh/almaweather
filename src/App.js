import React, { Component } from 'react';
import './App.css';
import { SearchField } from './SearchField';
import { WeatherRetriever } from "./WeatherRetriever";
import Cookies from 'universal-cookie';

class App extends Component {
    constructor(props) {
        super(props);

        const cookies = new Cookies();
        let savedPlaces = [];
        console.log(cookies.get('savedPlaces'));
        if (cookies.get('savedPlaces')!==undefined) {
            savedPlaces = cookies.get('savedPlaces')
        }

        this.state = {
            place: '' ,
            cookies: cookies,
            savedPlaces: savedPlaces
        };

        this.updatePlace = this.updatePlace.bind(this);
        this.savePlace = this.savePlace.bind(this);
    }

    updatePlace(newPlace) {
        this.setState({
              place: newPlace
        });
    }

    savePlace(placeToSave) {// Adds new placeToSave to savedPlaces array & cookies if not yet included
        if(!this.state.savedPlaces.includes(placeToSave)) {
            const sp = [placeToSave, ...this.state.savedPlaces];
            this.setState({
                savedPlaces: sp
            });

            const current = new Date();
            const nextYear = new Date();
            nextYear.setFullYear(current.getFullYear() + 1);

            this.state.cookies.set('savedPlaces', sp, { path: '/', expires: nextYear });
            console.log(this.state.cookies.get('savedPlaces'));
        }
    }

    render() {
        return (
            <div className="App container">
                <SearchField onChange={this.updatePlace}/>
                <WeatherRetriever place={this.state.place} onPlaceSave={this.savePlace}/>
            </div>
        );
    }
}

export default App;
