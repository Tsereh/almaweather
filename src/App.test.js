import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from "enzyme";
import App from './App';

it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<App />, div);
    ReactDOM.unmountComponentAtNode(div);
});

it("should remove duplicate entries/add new entries", () => {
    let app = shallow(<App/>);

    app.instance().togglePlaceBookmark("Helsinki", false);
    app.instance().togglePlaceBookmark("Vantaa", false);
    app.instance().togglePlaceBookmark("Helsinki", true);
    app.instance().togglePlaceBookmark("Espoo", false);

    expect(app.state().savedPlaces).toEqual(["Espoo", "Vantaa"]);
});
