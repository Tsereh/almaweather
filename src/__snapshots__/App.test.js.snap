// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component matches the snapshot 1`] = `
<div
  className="App container"
>
  <div
    className="search-field input-group"
  >
    <input
      className="form-control"
      id="search-field"
      onChange={[Function]}
      placeholder="Search"
      type="text"
      value=""
    />
  </div>
  <div
    className="main-weather"
  >
    <p
      className="main-weather-oneliner"
    >
      Please, enter a place name in to a search field to see it's weather.
    </p>
  </div>
  <div
    className="bookmarked-places"
  >
    <button
      aria-label={null}
      className="bookmarked-place btn btn-primary"
      onClick={[Function]}
      type="button"
      value="Espoo"
    >
      Espoo
    </button>
    <button
      aria-label={null}
      className="bookmarked-place btn btn-primary"
      onClick={[Function]}
      type="button"
      value="Vantaa"
    >
      Vantaa
    </button>
  </div>
</div>
`;
